lib/ripper-tags/json_formatter.rb

```diff
@@ -16,19 +16,37 @@ def stream_format?
 
     def with_output
       super do |true_out|
-        buffer = []
-        yield buffer
+        @buffer ||= []
+        yield @buffer
+        @buffer.uniq!
```

Comment:
What does `uniq` do here?

Reply:
The goal was to prevent duplicates if we run with append over the same set of files (or at least some intersection of) that are already present in tags. Vim and Emacs formatters do uniq too - vim directly, emacs by throwing away old symbols as we encounter new ones with the same name and sourcefile.